### DISCUSSION TITLE
Why is monorepo setup needed/recommended?

### DISCUSSION THREAD

**Discussion by LunatiqueCoder** [tamagui/tamagui#475] [Q&A] [answered]
Hello! Happy New Year everyone and thank you so much for the awesome Christmas gift of version 1.0! 

As the title says, I've got one question: **Why is monorepo setup needed/recommended?**

It seems to me that it complicates things too much by separating those concerns and I think expo's approach (of running iOS android and web from the same root) is much easier. What would be the caveats?

Thank you very much!

**Comment by sohcah** [comment, marked as answer]
I can confirm that a monorepo setup isn't _needed_, even if it is recommended.

I've got Tamagui running as just a single package (although I actually have it within a monorepo, the other packages aren't related to Tamagui/the app) using Vite for Web and Expo for Native.
I'm hoping to make a proper template for this at some point, but in the meantime here is a repo link: https://github.com/CuppaZee/CZ3/tree/main/app

**Reply by amoghkulkarnifr** [reply]
Hello @sohcah, we are trying to do something similar to what you have mentioned (Vite for web, Expo for Native) and facing a few issues while setting up. I was hoping to get some pointers from the repo link that you've given, but the codebase seems to have moved since. Would it be possible for you to give me the link to the new repo? Hoping that the new codebase is still public!  

Thanks a lot, in advance!
